Packs/GreyNoise/Integrations/GreyNoise/GreyNoise.py

```diff
@@ -78,54 +90,71 @@
-    'UNAUTHENTICATED': 'Unauthenticated. Check the configured API Key.',
-    'COMMAND_FAIL': 'Failed to execute {} command.\n Error: {}',
-    'SERVER_ERROR': 'The server encountered an internal error for GreyNoise and was unable to complete your request.',
-    'CONNECTION_TIMEOUT': 'Connection timed out. Check your network connectivity.',
-    'PROXY': "Proxy Error - cannot connect to proxy. Either try clearing the "
-             "'Use system proxy' check-box or check the host, "
-             "authentication details and connection details for the proxy.",
-    'INVALID_RESPONSE': 'Invalid response from GreyNoise. Response: {}',
-    'QUERY_STATS_RESPONSE': 'GreyNoise request failed. Reason: {}'
+    "API_RATE_LIMIT": "API Rate limit hit. Try after sometime.",
+    "UNAUTHENTICATED": "Unauthenticated. Check the configured API Key.",
+    "COMMAND_FAIL": "Failed to execute {} command.\n Error: {}",
+    "SERVER_ERROR": "The server encountered an internal error for GreyNoise and was unable to complete your request.",
+    "CONNECTION_TIMEOUT": "Connection timed out. Check your network connectivity.",
+    "PROXY": "Proxy Error - cannot connect to proxy. Either try clearing the "
+    "'Use system proxy' check-box or check the host, "
+    "authentication details and connection details for the proxy.",
+    "INVALID_RESPONSE": "Invalid response from GreyNoise. Response: {}",
+    "QUERY_STATS_RESPONSE": "GreyNoise request failed. Reason: {}",
 }
 
 
-''' CLIENT CLASS '''
+""" CLIENT CLASS """
 
 
 class Client(GreyNoise):
-    """Client class to interact with the service API
-    """
+    """Client class to interact with the service API"""
+
     def authenticate(self):
         """
         Used to authenticate GreyNoise credentials.
         """
-        response = str(self.test_connection())
-        if response == "Success: Access and API Key Valid":
-            return 'ok'
+        current_date = datetime.now()
+        try:
+            response = self.test_connection()
+            expiration_date = datetime.strptime(response["expiration"], "%Y-%m-%d")
+            if current_date < expiration_date and response["offering"] != "community":
+                return "ok"
+            else:
+                raise DemistoException(
+                    f"Invalid API Offering ({response['offering']})or Expiration Date ({expiration_date})"
+                )
 
-        code, body = parse_code_and_body(response)
-        if code and body:
-            raise exceptions.RequestFailure(code, body)
-        elif response == "":
-            raise exceptions.RateLimitError()
-        elif "ProxyError" in response:
-            raise requests.exceptions.ProxyError()
-        elif "timed out" in response:
-            raise requests.exceptions.ConnectTimeout()
-        else:
-            raise Exception(response)
+        except RateLimitError:
+            raise DemistoException(EXCEPTION_MESSAGES["API_RATE_LIMIT"])
```

Comment:
If this error is frequently seen, maybe you should consider implementing a retry mechanism